File: behaviorial/state/typescript/context.ts
import { StoppedState } from './concrete-state';
import { State } from './interface';

export class AudioPlayer {
	private state: State;

	constructor() {
		this.state = new StoppedState(); // Initial state
	}

	setState(state: State): void {
		this.state = state;
	}

	play(): void {
		this.state.play(this);
	}

	pause(): void {
		this.state.pause(this);
	}

	stop(): void {
		this.state.stop(this);
	}
}

File: behaviorial/state/python/context.py
from concrete_state import StoppedState


class AudioPlayer:
    def __init__(self):
        self.state = StoppedState()

    def set_state(self, state):
        self.state = state

    def play(self):
        self.state.play(self)

    def pause(self):
        self.state.pause(self)

    def stop(self):
        self.state.stop(self)
